### DISCUSSION TITLE
Predicting new topics

### DISCUSSION THREAD

**Discussion by VenkataBuddha** [MaartenGr/BERTopic#927] [Q&A] [answered]
Hi Everyone,
So when we use the model.transform(newdocs), we get the topics that were already generated/identifed in the pre trained model.
But what if we have a new topic in the newdocs. What can be done to assign a new topic to these new docs instead of them being assigned '-1'.

**Comment by MaartenGr** [comment, marked as answer]
Just to be sure I understand your question correctly, do you mean with predicting new topics that they fall in an outlier category, that they fall in a previously found topic, or that they may end up in an entirely new topic?

If it's about going from an outlier (-1) to a non-outlier topic, then your can use the function [.reduce_outliers](https://maartengr.github.io/BERTopic/getting_started/outlier_reduction/outlier_reduction.html) for that.

If it's about discovering entirely new topics, then it might be worthwhile to use [online BERTopic](https://maartengr.github.io/BERTopic/getting_started/online/online.html) for that which, combined with River, allows for discovering topics that were not found before.

**Reply by VenkataBuddha** [reply]
Thank You for the quick response Maarten!
Apologies for the confusion in my post, but yes I mean about docs ending up in entirely new topics.
I will reffer the online BERTopic documentation!
